### PR TITLE
Add print_key.py

### LINKED ISSUE #47: Update key/list.sh
https://github.com/EasyGnuPG/pgpg/blob/d108baf68dc2eb6de3d360b4bb1e06f2017d0946/src/cmd/key/list.sh#L38

https://github.com/EasyGnuPG/pgpg/blob/d108baf68dc2eb6de3d360b4bb1e06f2017d0946/src/cmd/key/list.sh#L42

some scripts from #28 #40 can be same

### PR DESCRIPTION
**Pull request by diveshuttam**
closes #47

### CODE REVIEW

**Review by dashohoxha — DISMISSED** [review]
I have not tested it, but it seems to be ok.

**Comment by diveshuttam** [comment]
@sourabhtk37 I would like you to have a look at the python code once before merging. Please review whenever you are free. We will merge after that.
I have taken care of PEP8 but still I feel there might be some ways to improve the code which I might have overlooked.

File: src/gpg/fn/print_key.py
Comment (sourabhtk37) [inline src/gpg/fn/print_key.py]: Empty print block. Use the `print` functions `end` parameter to specify newline.
`print(key, end="\n")`

File: src/gpg/fn/print_key.py
Comment (sourabhtk37) [inline src/gpg/fn/print_key.py]: Keep formatting content together where it's used, for ease of editing. 
Move line 28,29,32 to 91.

File: src/gpg/fn/print_key.py
Comment (sourabhtk37) [inline src/gpg/fn/print_key.py:41]: I see only 4 levels of trust here: https://www.gnupg.org/gph/en/manual/x334.html
Can you specify the source?

File: src/gpg/fn/print_key.py
Comment (sourabhtk37) [inline src/gpg/fn/print_key.py]: Unnecessary variables here: `subkey_id`, and `tformat`.

File: src/gpg/fn/print_key.py
Comment (sourabhtk37) [inline src/gpg/fn/print_key.py]: Can simplify this if block:
```python
if sign.keyid != identity:
    if 0 in (sign.revoked, sign.expired):

File: src/gpg/fn/print_key.py
Comment (sourabhtk37) [inline src/gpg/fn/print_key.py]: Simply do:
```python
if subkey.can_sign:

**Review by sourabhtk37 — DISMISSED** [review]

File: src/gpg/fn/print_key.py
Comment (diveshuttam) [reply src/gpg/fn/print_key.py:41]: I used `help(gpg.constants.validity)` in the python shell to find all of these. You can find them in the c library also [here](https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gpgme.git;a=blob;f=src/gpgme.h.in;hb=d8beab30c44482fb7a3e445b92cec482792b2ca0#l310) highlighted below:
```c
/* The available validities for a trust item or key.  */
 311 typedef enum
 312   {
 313     GPGME_VALIDITY_UNKNOWN   = 0,
 314     GPGME_VALIDITY_UNDEFINED = 1,
 315     GPGME_VALIDITY_NEVER     = 2,
 316     GPGME_VALIDITY_MARGINAL  = 3,
 317     GPGME_VALIDITY_FULL      = 4,
 318     GPGME_VALIDITY_ULTIMATE  = 5
 319   }
 320 gpgme_validity_t;
```
I initially found these by looking at python tests though they don't contain all. you can look them up [here](https://git.gnupg.org/cgi-bin/gitweb.cgi?p=gpgme.git&a=search&h=HEAD&st=grep&s=constants.validity.)
I have mapped them to the ones listed above in the url you mentioned and since undefined was extra, I mapped it to unknown only. UNDEFINED would have been treated as UNKNOWN in existing bash script also. This is also compatible with `egpg contact certify`.

Comment (dashohoxha) [reply src/gpg/fn/print_key.py:41]: I understand that ULTIMATE is a trust level that you assign only to your own keys. However it is not clear to me the difference between UNKNOWN and UNDEFINED. Maybe UNDEFINED is the initial value (which means no-value), but you cannot assign UNDEFINED, you can only assign UNKNOWN.

This can be a good question to ask on gnupg-devel. But maybe this issue is not so important, after all.

**Comment by diveshuttam** [comment]
@sourabhtk37 can you review this again and if all's OK please approve this.

**Review by sourabhtk37 — APPROVED** [review]
Seems Ok.

**Review by dashohoxha — APPROVED** [review]
You can merge it for the time being, and we can always make changes/improvements later if needed.

**Pull request merged**
